plugins/qiskit_executor/routes.py

```diff
@@ -213,48 +211,49 @@ def post(self, arguments: CircuitParameters):
-                db_id=db_task.id,
-                _external=True,
-            )
-            ui_href = url_for(
-                f"{QISKIT_EXECUTOR_BLP.name}.AuthenticationFrontend",
-                db_id=db_task.id,
-                _external=True,
-            )
-            db_task.progress_target = 2
-            db_task.add_next_step(href=href, ui_href=ui_href, step_id="authentication")
-            db_task.add_task_log_entry("Starting authentication step")
-        elif not arguments.backend:
-            # start the backend selection task
-            href = url_for(
-                f"{QISKIT_EXECUTOR_BLP.name}.BackendSelectionView",
-                db_id=db_task.id,
-                _external=True,
-            )
-            ui_href = url_for(
-                f"{QISKIT_EXECUTOR_BLP.name}.BackendSelectionFrontend",
-                db_id=db_task.id,
-                _external=True,
-            )
-            db_task.progress_target = 2
-            db_task.add_next_step(href=href, ui_href=ui_href, step_id="backend-selection")
-            db_task.add_task_log_entry("Starting backend selection step")
-        else:
-            # start the execution task
-            task: chain = (
-                start_execution.s(db_id=db_task.id)
-                | result_watcher.si(db_id=db_task.id)
-                | save_task_result.s(db_id=db_task.id)
-            )
-            # save errors to db
-            task.link_error(save_task_error.s(db_id=db_task.id))
-            task.apply_async()
-
-        db_task.save(commit=True)
+        # store urls for celery task
+        auth_href = url_for(
+            f"{QISKIT_EXECUTOR_BLP.name}.AuthenticationView",
+            db_id=db_task.id,
+            _external=True,
+        )
+        auth_ui_href = url_for(
+            f"{QISKIT_EXECUTOR_BLP.name}.AuthenticationFrontend",
+            db_id=db_task.id,
+            _external=True,
+        )
+        backend_href = url_for(
+            f"{QISKIT_EXECUTOR_BLP.name}.BackendSelectionView",
+            db_id=db_task.id,
+            _external=True,
+        )
+        backend_ui_href = url_for(
+            f"{QISKIT_EXECUTOR_BLP.name}.BackendSelectionFrontend",
+            db_id=db_task.id,
+            _external=True,
+        )
+        PluginState.set_value(
+            QiskitExecutor.instance.identifier,
+            f"authentication_href_{db_task.id}",
+            auth_href,
+        )
+        PluginState.set_value(
+            QiskitExecutor.instance.identifier,
+            f"authentication_ui_href_{db_task.id}",
+            auth_ui_href,
+        )
+        PluginState.set_value(
+            QiskitExecutor.instance.identifier,
+            f"backend_selection_href_{db_task.id}",
+            backend_href,
+        )
+        PluginState.set_value(
+            QiskitExecutor.instance.identifier,
+            f"backend_selection_ui_href_{db_task.id}",
+            backend_ui_href,
+            commit=True,
+        )
```

Comment:
Instead of using the plugin state table (which is intended for global, not task specific state), you can just use the data attribute of the processing task to store small key value pairs (which you already do for parameters and options).

As a side note: You dump the parameters into a string, but the `data` field can hold arbitrary json serializable data, so there is no need to serialize json data into a string there. Only if you use the `parameters` attribute of the processing task directly (on the same level as the data attribute) do you need to use a string as value.